copier/config/user_data.py

```diff
@@ -108,8 +112,36 @@ def cast_answer_type(answer: Any, type_fn: Callable) -> Any:
         return answer
 
 
+def render_string(template: Any, env: SandboxedEnvironment, context: AnyByStrDict):
+    """Render a templated-string using Jinja."""
+    try:
+        return env.from_string(template).render(**context)
+    except TypeError:
+        # template was not a string
+        return template
+    except UndefinedError as error:
+        raise UserMessageError(str(error)) from error
```

Comment:
> and the returned result would be typed as expected

But what is *expected* here? If the `type` is `str` and the `default` is `"3"`, we expect a string, not an integer. In that case it's still fine because the answer will be cast again as `str`. But if the type is not set, I think the user still expects the default to be a string and not an integer, and the default `yaml` type will leave it as an integer.

Reply:
Hmm I see, you're right.

Then I guess the best you can do is modify the method name. Calling it `render_string` and expecting things that are not strings is confusing to the reader.

Maybe something like `render_if_string`? :thinking: 

Also, since all the type casting is indeed done later, I guess all of the comments about using `NativeEnvironment` should be ignored. Let me resolve them too.

Reply:
I agree that `render_string` is confusing! Maybe `render_value` then, since we're trying to render the *values* of the YAML keys? Otherwise `render_if_string` is fine as well!

Reply:
> Maybe `render_value`

:ok_hand:

Reply:
Done :slightly_smiling_face: